streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -777,4 +778,23 @@ public synchronized TopologyDescription describe() {
         return internalTopologyBuilder.describe();
     }
 
+    // Adjust the generated topology based on the configs.
+    // Not exposed as public API and should be removed post 2.0
+    Topology adjust(final StreamsConfig config) {
+        final boolean enableOptimization20 = config.getString(StreamsConfig.TOPOLOGY_OPTIMIZATION).equals(StreamsConfig.OPTIMIZE_AT_20);
```

Comment:
revert the check to avoid NPE: `StreamsConfig.OPTIMIZE_AT_20.equals(config.getString(StreamsConfig.TOPOLOGY_OPTIMIZATION))`

Reply:
I'm not sure I'm reading this right... If I say `optimize:=all`, we won't do this optimization, right? It seems like you want to check for `2.0` *or* `all`

Reply:
ack